### PR TITLE
Create RavenDB Gateway 1 to 2 upgrade guide

### PR DESCRIPTION
**Pull request by mauroservienti**

### CODE REVIEW

File: persistence/upgrades/ravendb-gateway-1to2.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+var ravenGatewayDeduplicationConfiguration = new RavenGatewayDeduplicationConfiguration((builder, _) => 
+{
+   //create and configure the RavenDB document store.
+})
+{
+   EnableClusterWideTransactions = true
+};
+```
+
+If the persister is configured to connect to a database, whose replication factor is greater than one, and cluster-wide transaction are not enabled, the persister will thow the following exception:
```
Comment (danielmarbach) [inline persistence/upgrades/ravendb-gateway-1to2.md:28]:
```suggestion
If the persister is configured to connect to a database, whose replication factor is greater than one, and cluster-wide transaction are not enabled, the persister will throw the following exception:
```

File: persistence/upgrades/ravendb-gateway-1to2.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+- nservicebus/gateway/ravendb
+reviewed: 2021-11-25
+isUpgradeGuide: true
+upgradeGuideCoreVersions:
+ - 7
+---
+
+## Cluster-wide transactions support
+
+Starting with NServiceBus.Gateway.RavenDB version 2, the gateway persistence supports RavenDB cluster-wide transactions. RavenDB 5.2 or greater is required for both the RavenDB client and server. Cluster-wide transactions support is disabled by default. To enable it, configure the persister like follows:
```
Comment (danielmarbach) [inline persistence/upgrades/ravendb-gateway-1to2.md:16]:
```suggestion
NServiceBus.Gateway.RavenDB version 2 supports cluster-wide transactions. RavenDB 5.2 or greater is required for both the RavenDB client and server. Cluster-wide transactions support is disabled by default. To enable it, configure the persister like the following:
```

File: persistence/upgrades/ravendb-gateway-1to2.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+```
```
Comment (danielmarbach) [inline persistence/upgrades/ravendb-gateway-1to2.md:18]: This should probably be a snippet no?

Comment (mauroservienti) [reply persistence/upgrades/ravendb-gateway-1to2.md:18]: We don't usually do snippets in upgrade guides, but in this case, it's an easy one.

Comment (danielmarbach) [reply persistence/upgrades/ravendb-gateway-1to2.md:18]: That is new to me. Maybe I've only seen old examples like https://github.com/Particular/docs.particular.net/blob/master/Snippets/ASB/ASB_9/Upgrade7to8.cs. Then don't bother Mauro.

**Review by danielmarbach — APPROVED** [review]

**Pull request merged**